### PR TITLE
Add number of splits and threads to printPlanWithStats

### PR DESCRIPTION
**Pull request by majetideepak**
Add split and driver counts to OperatorStats and PlanNodeStats.

Extend the output of printPlanWithStats to include the number of splits for leaf
nodes and the number of threads (drivers used by the pipeline). The number of
threads is excluded for plan nodes that contain multiple operators. 

Also, print input stats for all operators that belong to plan nodes with
multiple operators.

Sample partial output for q18 without custom stats.

```
-> LocalPartition[]
       Output: 1248 rows (145.60MB), Cpu time: 500.19ms, Blocked wall time: 6.10s, Peak memory: 0B
       LocalPartition: Input: 624 rows (72.77MB), Output: 624 rows (72.77MB), Cpu time: 498.21ms, Blocked wall time: 62.29ms, Peak memory: 0B, Threads: 8
       LocalExchangeSource: Input: 624 rows (72.83MB), Output: 624 rows (72.83MB), Cpu time: 1.98ms, Blocked wall time: 6.03s, Peak memory: 0B, Threads: 1
      -> HashJoin[INNER o_custkey=c_custkey]
         Output: 624 rows (72.77MB), Cpu time: 950.41ms, Blocked wall time: 2.90s, Peak memory: 37.00MB
         HashBuild: Input: 1500000 rows (0B), Output: 0 rows (0B), Cpu time: 689.25ms, Blocked wall time: 511.59ms, Peak memory: 36.00MB, Threads: 8
         HashProbe: Input: 624 rows (48.38MB), Output: 624 rows (72.77MB), Cpu time: 261.16ms, Blocked wall time: 2.39s, Peak memory: 1.00MB, Threads: 8
        -> HashJoin[INNER o_orderkey=l_orderkey]
           Output: 624 rows (48.38MB), Cpu time: 18.29ms, Blocked wall time: 44.45s, Peak memory: 2.00MB
           HashBuild: Input: 624 rows (14.03KB), Output: 0 rows (0B), Cpu time: 1.34ms, Blocked wall time: 0ns, Peak memory: 1.00MB, Threads: 1
           HashProbe: Input: 80619 rows (48.33MB), Output: 624 rows (48.38MB), Cpu time: 16.95ms, Blocked wall time: 44.45s, Peak memory: 1.00MB, Threads: 8
          -> TableScan[Table: orders]
             Input: 80619 rows (48.33MB), Raw Input: 14861668 rows (224.66MB), Output: 80619 rows (48.33MB), Cpu time: 560.08ms, Blocked wall time: 81.00us, Peak memory: 24.00MB, Threads: 8, Splits: 80
```

### CODE REVIEW

File: velox/exec/PlanNodeStats.cpp [hunk 1]
```diff
@@ -70,6 +78,14 @@ std::string PlanNodeStats::toString(bool includeInputStats) const {
       << ", Cpu time: " << succinctNanos(cpuWallTiming.cpuNanos)
       << ", Blocked wall time: " << succinctNanos(blockedWallNanos)
       << ", Peak memory: " << succinctBytes(peakMemoryBytes);
+
+  if (!driverIds.empty()) {
+    out << ", Threads: " << driverIds.size();
+  }
+
+  if (numSplits > 0) {
+    out << ", NumSplits: " << numSplits;
```
Comment (mbasmanova) [inline velox/exec/PlanNodeStats.cpp:87]: NumSplits -> Splits

File: velox/exec/PlanNodeStats.h [hunk 2]
```diff
@@ -84,11 +84,21 @@ struct PlanNodeStats {
   /// Breakdown of stats by operator type.
   std::unordered_map<std::string, std::unique_ptr<PlanNodeStats>> operatorStats;
 
+  /// Drivers that executed the pipeline.
+  std::set<int> driverIds;
```
Comment (mbasmanova) [inline velox/exec/PlanNodeStats.h:88]: std::unordered_set

File: velox/exec/Operator.h [hunk 3]
```diff
@@ -129,6 +129,7 @@ struct OperatorStats {
   MemoryStats memoryStats;
 
   std::unordered_map<std::string, RuntimeMetric> runtimeStats;
+  std::set<int> driverIds;
```
Comment (mbasmanova) [inline velox/exec/Operator.h:132]: - std::unordered_set
- maintaining a set of driver IDs is a lot more expensive than maintaining a simple counter; any particular reason not to use a simpler counter?

File: velox/exec/PlanNodeStats.cpp [hunk 4]
```diff
@@ -53,14 +52,23 @@ void PlanNodeStats::addTotals(const OperatorStats& stats) {
   for (const auto& entry : stats.runtimeStats) {
     customStats[entry.first].merge(entry.second);
   }
+  // Populating drivers for plan nodes with multiple operators is not useful.
+  // Each operator could have been executed in different pipelines with
+  // different number of drivers.
+  if (!isMultiOperatorNode()) {
+    driverIds.insert(stats.driverIds.begin(), stats.driverIds.end());
```
Comment (mbasmanova) [inline velox/exec/PlanNodeStats.cpp:59]: this can be just driverIds = stats.driverIds

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Looks good % some questions and comments below. It is useful to see the number of splits and threads in the output of printPlanWithStats. Thanks for adding this information.

Try to use active verbs in PR title and description for clarity.

File: velox/exec/Operator.h (hunk 3 above)
Comment (majetideepak) [reply velox/exec/Operator.h]: Fixed to use a simpler counter. Each driver updates the stats on close, so a counter should work.

**Comment by majetideepak** [comment]
@mbasmanova thanks for improving the title and description. Look much better.

File: velox/exec/Operator.h [hunk 5]
```diff
@@ -130,6 +130,8 @@ struct OperatorStats {
 
   std::unordered_map<std::string, RuntimeMetric> runtimeStats;
 
+  int driverCount = 0;
```
Comment (mbasmanova) [inline velox/exec/Operator.h:133]: drivers or numDrivers for consistency

File: velox/exec/PlanNodeStats.h [hunk 6]
```diff
@@ -84,11 +84,21 @@ struct PlanNodeStats {
   /// Breakdown of stats by operator type.
   std::unordered_map<std::string, std::unique_ptr<PlanNodeStats>> operatorStats;
 
+  /// Number of drivers that executed the pipeline.
+  int driverCount{0};
```
Comment (mbasmanova) [inline velox/exec/PlanNodeStats.h:88]: drivers or numDrivers for consistency

**Review by mbasmanova — COMMENTED** [review]
Looks great % minor comment.

**Review by mbasmanova — APPROVED** [review]
@majetideepak Thank you, Deepak.

**Pull request closed without merging**